benchmarks/benchmark_filters.py

```diff
@@ -12,6 +12,35 @@ def setup(self):
         self.image[3000:, 3000] += 0.5
 
     def time_sobel(self):
-        result = filters.sobel(self.image)
+        filters.sobel(self.image)
 
 
+class MultiOtsu(object):
+    """Benchmarks for MultiOtsu threshold."""
+    param_names = ['classes']
+    params = [3, 5]
+
+    def setup(self, *args):
+        self.image = data.camera()
+
+    def time_threshold_multiotsu(self, classes):
+        filters.threshold_multiotsu(self.image, classes=classes)
+
+    def peakmem_reference(self, *args):
+        """Provide reference for memory measurement with empty benchmark.
+
+        Peakmem benchmarks measure the maximum amount of RAM used by a
+        function. However, this maximum also includes the memory used
+        during the setup routine (as of asv 0.2.1; see [1]_).
+        Measuring an empty peakmem function might allow us to disambiguate
+        between the memory used by setup and the memory used by target (see
+        other ``peakmem_`` functions below).
+
+        References
+        ----------
+        .. [1]: https://asv.readthedocs.io/en/stable/writing_benchmarks.html#peak-memory
+        """
+        pass
+
+    def peakmem_threshold_multiotsu(self, classes):
+        filters.threshold_multiotsu(self.image, classes=classes)
```

Comment:
Very nice! Could you show the results of `asv dev` on this? I'm just curious to see how it compares to your earlier attempts.

Reply:
This version is way better:
```
[ 18.75%] ··· benchmark_filters.MultiOtsu.peakmem_reference                  ok
[ 18.75%] ··· ========= =======
               classes         
              --------- -------
                  3      71.4M 
                  4      71.4M 
                  5      71.4M 
              ========= =======

[ 20.83%] ··· ...lters.MultiOtsu.peakmem_threshold_multiotsu                 ok
[ 20.83%] ··· ========= =======
               classes         
              --------- -------
                  3      74.7M 
                  4      74.7M 
                  5      74.7M 
              ========= =======

[ 22.92%] ··· ..._filters.MultiOtsu.time_threshold_multiotsu                 ok
[ 22.92%] ··· ========= ==========
               classes            
              --------- ----------
                  3      82.7±0ms 
                  4      148±0ms  
                  5      1.38±0s  
              ========= ==========
```

Reply:
@alexandrejaguar awesome!

Watch that exponential explode LOL. But yes, even with 5 classes it's pretty damn good! I'm so happy. =)

Reply:
@jni I was testing in some larger images (2000 X 2000), and it goes to around 1 minute with six classes!

Reply:
I think Stéfan suggested to put the complexity in the docs somewhere. I think it is approximated by (for the brute forcing step, not the preparation):
```
combinations = nbins**(classes-1)/(classes-1)!
variance_calc = classes + 1
complexity = O(combinations * variance_calc)
```